### PR TITLE
remove the syslog lambda custom name so the function can be replaced

### PR DESCRIPTION
**Pull request by mattmanning**
When existing syslog resources are linked or unlinked Convox will try to replace them and create a new function in the nodejs 4.3 Lambda runtime because of the nodejs 0.10 Lambda runtime deprecation.

That replacement currently fails because the function has a custom name. This change removes the custom name.

This change is dependent on https://github.com/convox/syslog/pull/6 and should not be released until the new lambda.zip has been uploaded to our S3 buckets.

### CODE REVIEW

File: provider/aws/templates/service/syslog.tmpl [hunk 1]
```diff
@@ -200,6 +197,7 @@
                         {
                           "Ref": "AWS::StackName"
                         },
+                        "*",
```
Comment (ddollar) [inline provider/aws/templates/service/syslog.tmpl]: Should probably be -* to avoid spillage from e.g.  `dev` and `dev-foo`

Comment (mattmanning) [inline provider/aws/templates/service/syslog.tmpl]: Good point, thanks

**Comment by mattmanning** [comment]
[Handle custom-name replacement errors for syslog Lambda](https://trello.com/c/pfKyfj6s/1537-handle-custom-name-replacement-errors-for-syslog-lambda)

File: provider/aws/templates/service/syslog.tmpl (hunk 1 above)
Comment (ddollar) [inline provider/aws/templates/service/syslog.tmpl]: Should actually be `-Function-*`

**Comment by ddollar** [comment]
Seems like this might have gotten buried but the wildcard should be `-Function-*`

**Comment by mattmanning** [comment]
My bad. I misread your original comment. Pattern should be correct now.

**Comment by ddollar** [comment]
Needs rebase, removing my review until then.

**Comment by mattmanning** [comment]
@ddollar Just noticed this one. Rebased and `make test` is passing.

**Comment by mattmanning** [comment]
@ddollar Just rebased again. This one should be good to go.

**Pull request closed without merging**